src/WooCommerce.php

```diff
@@ -342,18 +342,16 @@ public static function woocommerce_template_loop_product_thumbnail() {
 
       if (count($variation_ids)) {
         $placeholders = implode(',', array_fill(0, count($variation_ids), '%d'));
-        $attachment_ids = array_unique(
```

Comment:
The new query looks good, but we still need the array_unique() here to avoid the product's primary image to be repeated, in case a variant uses the same image.

Alternatively, we can omit the array_unique() and instead filter out the primary image ID we already have in the query — see below.